x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/rules_table_context.test.tsx

```diff
@@ -23,22 +25,51 @@ import { useRulesTableSavedState } from './use_rules_table_saved_state';
 
 jest.mock('../../../../../common/lib/kibana');
 jest.mock('../../../../rule_management/logic/use_find_rules');
+jest.mock('../../../../rule_management/api/hooks/use_fetch_rules_snooze_settings');
 jest.mock('./use_rules_table_saved_state');
 
-function renderUseRulesTableContext(
-  savedState: ReturnType<typeof useRulesTableSavedState>
-): RulesTableState {
+function renderUseRulesTableContext({
+  rules,
+  rulesSnoozeSettings,
+  savedState,
+}: {
+  rules?: Rule[] | Error;
+  rulesSnoozeSettings?: RuleSnoozeSettings[] | Error;
+  savedState?: ReturnType<typeof useRulesTableSavedState>;
+}): RulesTableState {
   (useFindRules as jest.Mock).mockReturnValue({
-    data: { rules: [], total: 0 },
+    data: rules instanceof Error || !rules ? undefined : { rules, total: rules?.length },
     refetch: jest.fn(),
     dataUpdatedAt: 0,
-    isFetched: false,
-    isFetching: false,
-    isLoading: false,
+    isFetched: !!rules,
+    isFetching: !rules,
+    isLoading: !rules,
     isRefetching: false,
+    isError: rules instanceof Error,
+  });
+  (useFetchRulesSnoozeSettings as jest.Mock).mockReturnValue({
+    data: rulesSnoozeSettings instanceof Error ? undefined : rulesSnoozeSettings,
+    isError: rulesSnoozeSettings instanceof Error,
```

Comment:
This function is used locally for convenience and accepts the state it returns as parameters. Instead of values an error can be passed to mimic loading failure in a simple way. I'm not sure if I get the point of refactoring here. Let's discuss it offline.